database/database_handler.py

```diff
@@ -111,3 +93,48 @@ def managed_session(self):
             raise e
         finally:
             session.close()
+
+    def close_connection(self):
+        try:
+            # Dispose the engine
+            self.engine.dispose()
+        except SQLAlchemyError as e:
+            LOGGER.info(f"Error disposing the database engine: {str(e)}")
+            raise e
+
+    @staticmethod
+    def extract_upload_date(path):
+        parts = path.split('/')
+        image_filename = parts[-1]
+        date_time_str = parts[-2]
+
+        try:
+            image_upload_date = datetime.strptime(date_time_str, "%Y-%m-%d_%H:%M:%S")
+            image_upload_date = image_upload_date.strftime("%Y-%m-%d")
+        except ValueError as e:
+            LOGGER.info(f"Invalid folder structure, can not retrieve date: {date_time_str}")
+            raise e
+
+        return image_filename, image_upload_date
+
+    def _renew_token_periodically(self):
+        while True:
+            # Calculate the time to renew the token
+            renew_time = self.token_expiration_time - self.token_renewal_margin
+
+            # Sleep until the renewal time
+            sleep_time = (renew_time - datetime.now()).total_seconds()
+            if sleep_time > 0:
+                LOGGER.info(f"Sleeping for {sleep_time} seconds until token renewal.")
+                time.sleep(sleep_time)
+
+            # Renew the token after the sleep
+            self._get_db_access_token()
+            LOGGER.info("Token for database renewed.")
+
+    def start_token_renewal_thread(self):
+        # TODO will this have conflicts with the threads used by YOLOv5?
```

Comment:
Im not able to see the monitoring tab in Azure ML, I want to check the memory and CPU or GPU  usage